### PR TITLE
Main Python integration doc

### PR DESCRIPTION
**Pull request by genzgd**

### CODE REVIEW

File: docs/en/integrations/language-clients/python/index.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
+---
+sidebar_label: ClickHouse Connect
+sidebar_position: 1
+keywords: [clickhouse, python, client, connect, integrate]
+slug: /en/integrations/python
+description: A suite of Python packages for connecting Python to ClickHouse
+---
+
+# ClickHouse Connect
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:9]: Could you add a link to this file as done in https://github.com/ClickHouse/clickhouse-docs/pull/412/?

File: docs/en/integrations/language-clients/python/index.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+## Introduction
+ClickHouse Connect is a suite of packages providing interoperability with a wide range of Python applications.
+The three primary components are:
+
+- A low level driver in the package `clickhouse_connect.driver`.  This package includes a basic client that handles 
+all ClickHouse Connect requests to the ClickHouse Server, as well assorted helper classes and utility functions.
+- A limited SQLAlchemy dialect in the package `clickhouse_connect.cc_sqlalchemy`.  This package focuses
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:17]: Should we add links to the `SQLAlchemy ` official docs?

File: docs/en/integrations/language-clients/python/index.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+## Requirements and Compatibility
+
+ClickHouse Connect requires Python 3.7 or greater.  It has been tested on Linux (x86 and Aarch64), Windows, and Mac OS
+(x86 and M1), but should run on most Python installations.
+
+ClickHouse Connect uses the ClickHouse HTTP protocol and has been tested against all currently supported ClickHouse
+versions.  It should also work correctly for most older versions of ClickHouse.
+
+The limited SQLAlchemy implementation is compatible with SQLAlchemy versions 1.3.x and 1.4.x.  The ClickHouse Connect
+Superset package is compatible with and has been tested against Superset versions 1.4.x, 1.5.x, and 2.0.0.
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:34]: Maybe we should create a table for quick scans?

File: docs/en/integrations/language-clients/python/index.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+
+## Installation
+
+Install ClickHouse Connect from PyPI via pip:
+
+`pip install clickhouse-connect`
+
+ClickHouse connect can also be installed from source by checking out the GitHub repository and running `pip install .`
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:42]: A link to the repository is required

File: docs/en/integrations/language-clients/python/index.md [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+
+`pip install clickhouse-connect`
+
+ClickHouse connect can also be installed from source by checking out the GitHub repository and running `pip install .` 
+in the project root directory.
+
+
+## Support Policy
+
+ClickHouse Connect is currently in beta and only the current beta release is supported.  Please update to the latest
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:48]: Shouldn't we list compatible ClickHouse versions?

File: docs/en/integrations/language-clients/python/index.md [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+| decompress        | Whether the data sent to ClickHouse server must be decompressed.  This setting is should only be used for "raw" requests.                                             |
+| quota_key         | The quota key associated with this requests.  See the ClickHouse server documentation on quotas.                                                                      |
+| session_check     | Used to check the session status.                                                                                                                                     |
+| session_timeout   | Number of seconds of inactivity before the identified by the session id will timeout and no longer be considered valid.  Defaults to 60 seconds.                      |
+| wait_end_of_query | Buffers the entire response on the ClickHouse server.  This setting is necessary to return summary information.  It is set automatically when *send_progress*=*True*. |
+
+All other keyword args and interpreted as ClickHouse user settings for each request. Please see the full ClickHouse  documentation for a complete list.  Note that
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:132]: Should we link https://clickhouse.com/docs/en/operations/settings/settings? There are many other settings available

File: docs/en/integrations/language-clients/python/index.md [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+client.query('SELECT * FROM some_table WHERE metric >= %s AND ip_address = %s', parameters=parameters)
+
+# Generates the following query:
+# SELECT * FROM some_table WHERE metric >= 35200.44 AND ip_address = '68.61.4.254''  
+```
+
+#### Settings Argument
+
+All the core SQL Client ClickHouse database server accept an optional `settings` keyword argument used for passing 
+ClickHouse Server users settings for the included SQL statement.  The `settings` argument should be a dictionary.
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:219]: Let's add another link to `settings` page

File: docs/en/integrations/language-clients/python/index.md [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+### Client Core SQL Methods 
+
+ClickHouse Connect provides four core methods for executing SQL statements against a ClickHouse database.
+
+- `raw_query` -- Execute any SQL statement and return the raw response as a byte array.  This bypasses any additional processing by the ClickHouse Connect driver.
+- `command` -- Execute a SQL command that does not return data
+- `query` -- Execute a SQL query and return a result set of Python values.  There are also specialized variants:
+  - `query_np` -- Execute a SQL query and return a Numpy array (Numpy installation required)
+  - `query_df` -- Execute a SQL query and return a Pandas Dataframe
+  - `query_arrow` -- Execute a SQL query and return a PyArrow Table
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:179]: Maybe it's worth adding `Supported Data formats` section?

File: docs/en/integrations/language-clients/python/index.md [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+| database             | str  | *None*             | The default database for the connection.  If not set, ClickHouse Connect will use the default database for *username*.                                                                                                                                                       |
+| compress             | bool | True               | Request gzip compression from ClickHouse HTTP requests.  Either the ClickHouse server must have the setting `enable_http_compression=1`, or the *username* must have permission to send settings with the request.                                                           |
+| query_limit          | int  | 5000               | Maximum number of rows to return for any `query` response.  Set this to zero to return unlimited rows.                                                                                                                                                                       |
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:100]: Do we need a warning it might cause an Out-of-memory exception?

**Comment by DanRoscigno** [comment]
@genzgd do you expect to add more files to the python dir?

File: docs/en/integrations/language-clients/python/index.md [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+- A Superset EngineSpec in the `clickhouse_connect.cc_superset`.  This package will automatically add a **ClickHouse
+Connect** Superset connector when ClickHouses Connet installed.  This EngineSpec supports all core Superset query
+functionality, but does not currently support certain advanced features such as file upload to a ClickHouse table.
+
+## Requirements and Compatibility
+
+ClickHouse Connect requires Python 3.7 or greater.  It has been tested on Linux (x86 and Aarch64), Windows, and Mac OS
+(x86 and M1), but should run on most Python installations.
+
+ClickHouse Connect uses the ClickHouse HTTP protocol and has been tested against all currently supported ClickHouse
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:30]: Shouldn't we list compatible ClickHouse versions?

File: docs/en/integrations/language-clients/python/index.md [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+
+## Core Driver API
+
+### Client Initialization
+
+The `clickhouse_connect.driver.client` class provides the primary interface between a Python application and the
+ClickHouse database server. Use the `clickhouse_connect.get_client` function to obtain a Client instance, which accepts
+the following parameters:
+
+#### Connection Parameters
```
Comment (mshustov) [inline docs/en/integrations/language-clients/python/index.md:90]: Should we add any details about the connection pool?

File: docs/en/integrations/language-clients/python/index.md (hunk 10 above)
Comment (genzgd) [reply docs/en/integrations/language-clients/python/index.md:30]: There are definitely some more detailed docs coming, but I don't think they'll end up on separate pages quite yet.

**Comment by genzgd** [comment]
@DanRoscigno

 I lied, I've decided to break it into separate pages.  There will be more pages coming over the next week or so.

**Review by DanRoscigno — APPROVED** [review]

**Pull request merged**